lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -412,6 +428,18 @@ instance Buildable Direction where
         Outgoing -> "outgoing"
         Incoming -> "incoming"
 
+instance FromText Direction where
+    fromText txt = case txt of
+        "outgoing" -> Right Outgoing
+        "incoming" -> Right Incoming
+        _ ->
+            Left . TextDecodingError $ show txt
+                   <> " is neither \"outgoing\", nor \"incoming\""
```

Comment:
https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-variable-length-indentation

Nit-picking, but one level of indent should be 4 characters (rather than 7).

Suggest:
```hs
            Left . TextDecodingError $ show txt
                 <> " is neither \"outgoing\", nor \"incoming\""
```